components/hab-spider/src/main.rs

```diff
@@ -0,0 +1,191 @@
+            .help("The path to the packages root")
+            .required(true)
+            .index(1))
+        .get_matches();
+
+    let path = matches.value_of("PATH").unwrap();
+
+    println!("Crawling packages... please wait.");
+
+    let mut spider = Spider::new(&path);
+    let start_time = PreciseTime::now();
+    let (ncount, ecount) = spider.crawl();
+    let end_time = PreciseTime::now();
+
+    println!("OK: {} nodes, {} edges ({} sec)",
+             ncount,
+             ecount,
+             start_time.to(end_time));
+
+    println!("\nAvailable commands: HELP, STATS, TOP, FIND, RDEPS, EXIT\n");
+
+    let mut done = false;
+    while !done {
+        print!("spider> ");
+        io::stdout().flush().ok().expect("Could not flush stdout");
+
+        let mut cmd = String::new();
+        io::stdin().read_line(&mut cmd).unwrap();
+
+        let v: Vec<&str> = cmd.trim_right().split_whitespace().collect();
+
+        if v.len() > 0 {
+            match v[0].to_lowercase().as_str() {
+                "help" => do_help(),
+                "stats" => do_stats(&spider),
+                "top" => {
+                    let count = if v.len() < 2 {
+                        10
+                    } else {
+                        v[1].parse::<usize>().unwrap()
+                    };
+                    do_top(&spider, count);
+                }
+                "find" => {
+                    if v.len() < 2 {
+                        println!("Missing search term\n")
+                    } else {
+                        let max = if v.len() > 2 {
+                            v[2].parse::<usize>().unwrap()
+                        } else {
+                            10
+                        };
+                        do_find(&spider, v[1].to_lowercase().as_str(), max)
+                    }
+                }
+                "rdeps" => {
+                    if v.len() < 2 {
+                        println!("Missing package name\n")
+                    } else {
+                        let max = if v.len() > 2 {
+                            v[2].parse::<usize>().unwrap()
+                        } else {
+                            10
+                        };
+                        do_rdeps(&spider, v[1].to_lowercase().as_str(), max)
+                    }
+                }
+                "exit" => done = true,
+                _ => println!("Unknown command\n"),
+            }
+        }
+    }
+}
+
+fn do_help() {
+    println!("HELP                  - print this message");
+    println!("STATS                 - print graph statistics");
+    println!("TOP [<count>]         - print nodes with the most reverse dependencies");
+    println!("FIND  <term> [<max>]  - find packages that match the search term, up to max items");
+    println!("RDEPS <name> [<max>]  - print the reverse dependencies for the package, up to max");
+    println!("EXIT                  - exit the application\n");
```

Comment:
No, tt's a good point, it would make sense to have the help output be stylistically similar to the other hab commands.